w3lib/url.py

```diff
@@ -94,7 +102,7 @@ def safe_url_string(url, encoding="utf8", path_encoding="utf8", quote_path=True)
     return urlunsplit(
         (
             parts.scheme,
-            netloc.rstrip(":"),
+            netloc.decode().rstrip(":"),
```

Comment:
I’m unsure about the approach in this function.

I assume the goal is to avoid `netloc` having a different type in different parts of the code. If so, what about, instead of making it `bytes`, make it `str` with a different variable in the `try except`?

```python
    try:
        netloc_bytes = parts.netloc.encode("idna")
    except UnicodeError:
        netloc = parts.netloc
    else:
        netloc = netloc_bytes.decode()
```